docs/contract/dataplanes-dataframes.md

```diff
@@ -0,0 +1,91 @@
+
+# Grafana data structure
+
+Grafana supports a variety of different data sources, each with its own data model. To make this possible, Grafana consolidates the query results from each of these data sources into one unified data structure called a **data frame**. The **data plane** is built on top of the data frame and includes additional information on what the data frame holds. 
+
+## Data frames overview  
+
+A data frame is a data structure that consolidates the query results from your data sources, providing a common container in Grafana. 
+
+:::caution
+
+Query responses are often more than one single data frame. 
+
+:::
+
+The data frame a column-oriented table (the _fields_) with metadata (the _frame_) attached. Since data frame columns can have labels attached to them (`key=value`, `key2=val`...), it can hold Prometheus like responses as well. 
+
+Each field in a data frame contains optional information about the values in the field, such as units, scaling, and so on. By adding field configurations to a data frame, Grafana can configure visualizations automatically. For example, you could configure Grafana to automatically set the unit provided by the data source.
+
+## Data planes overview  
+
+The data plane adds a property layer to the frame as metadata. It indicates the data frame _type_ (for example, a timeseries or a heatmap), which consists of a _kind_ (of data) and its _format_ (Prometheus-like, SQL-table-like). The use of data planes is generally not enforced, although it's mandatory for labeled data when using SQL expressions.
```

Comment:
I think if you are someone building a custom datasource plugin you might not be familiar with prometheus. To me this is not a "type" in the way sql is a type. Can we be more generic? Do we mean like timeseries data?